Casks/a/amneziavpn.rb

```diff
@@ -8,9 +8,20 @@
   desc "VPN client"
   homepage "https://amnezia.org/"
 
+  # Upstream tags GitHub release that can be pre-releases or betas,
+  # so we need to check the download page for the latest stable version.
+  # The website is hydrated with JavaScript, so we need to extract
+  # the version from the JavaScript file.
   livecheck do
-    url :url
-    strategy :github_latest
+    url "https://amnezia.org/en/downloads"
+    regex(/AmneziaVPN[._-]v?(\d+(?:\.\d+)+)\.dmg/i)
+    strategy :page_match do |page, regex|
+      js_file = page[%r{src=["']/assets/?(index.+\.js)\??["' >]}i, 1]
```

Comment:
```suggestion
      js_file = page[%r{src=["']?/assets/(index.+\.js)\??["' >]}i, 1]
```

It seems like the `?` was misplaced.